topdown/topdown_test.go

```diff
@@ -1465,6 +1465,84 @@ func TestTopDownTime(t *testing.T) {
 
 }
 
+func TestTopDownWalkBuiltin(t *testing.T) {
+
+	tests := []struct {
+		note     string
+		rules    []string
+		expected interface{}
+	}{
+		{
+			note: "scalar",
+			rules: []string{
+				`p[x] { walk(data.a[0], x) }`,
+			},
+			expected: `[
+				[[], 1]
+			]`,
+		},
+		{
+			note: "arrays",
+			rules: []string{
+				`p[x] { walk(data.a, x) }`,
+			},
```

Comment:
I should have asked "Walk() is a relation, instead of a function?"  I can't see any reason not to support builtins that are relations.  And a different syntax wouldn't change the fact that it's a relation.